### PR TITLE
Cleanup export

### PR DESCRIPTION
**Pull request by fsimonis**
## Main changes of this PR

This PR cleans-up the export mechanism by
* removing the unused feature to only export a specific type of export.
* renaming the VTKXML exporter to VTU

The PR also adds a serial exporter test to the VTUExporter.

## Motivation and additional information

General maintenance.

## Author's checklist

* [ ] I added a changelog file with this PR number in `docs/changelog/` if there are noteworthy changes.
* [x] I ran `tools/formatting/check-format` and everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.10.
* [x] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

File: src/io/tests/ExportVTUTest.cpp [hunk 1]
```diff
@@ -19,27 +19,32 @@ class Vertex;
...
+BOOST_AUTO_TEST_CASE(ExportPolygonalMeshSerial)
+{
+  PRECICE_TEST(""_on(1_rank).setupMasterSlaves());
+  int             dim = 2;
+  mesh::Mesh      mesh("MyMesh", dim, testing::nextMeshID());
+  mesh::Vertex &  v1      = mesh.createVertex(Eigen::VectorXd::Constant(dim, 0.0));
+  mesh::Vertex &  v2      = mesh.createVertex(Eigen::VectorXd::Constant(dim, 1.0));
+  Eigen::VectorXd coords3 = Eigen::VectorXd::Constant(dim, 0.0);
+  coords3(0)              = 1.0;
+  mesh::Vertex &v3        = mesh.createVertex(coords3);
```
Comment (MakisH) [inline src/io/tests/ExportVTUTest.cpp:37]: I understand that here you are constructing:
- v1 at (0.0, 0.0)
- v2 at (1.0, 1.0)
- v3 at (1.0, 0.0)

which, for `dim = 3`, would become:
- v1 at (0.0, 0.0, 0.0)
- v2 at (1.0, 1.0, 1.0)
- v3 at (1.0, 0.0, 0.0)

This confused me a bit, especially due to the name `coords3`, which is not really used further down (could be named, e.g. `v3_coords`). I assume it is not possible / good practice to move a `mesh::Vertex` after creating it?

Also, is line 37 properly formatted? Strange that the spacing is different.

**Review by MakisH — APPROVED** [review]
Looks good, always nice to simplify!

Still missing a changelog entry ("vtu now available for serial participants as well"), I guess this is on your radar.

I also added a point about vtu files in the exports documentation: https://github.com/precice/precice.github.io/blob/master/pages/docs/configuration/configuration-export.md

File: src/io/Export.hpp [hunk 2]
```diff
@@ -14,12 +14,9 @@ namespace io {
...
 class Export {
 public:
-  Export &operator=(Export &&) = delete;
-
-  virtual ~Export() {}
+  virtual ~Export() = default;
 
-  /// Returns the export type ID.
-  virtual int getType() const = 0;
+  Export &operator=(Export &&) = delete;
```
[the remark this answers is not in the document]
Comment (fsimonis) [reply src/io/Export.hpp:19]: Right, I guess we should follow the rule of 5 here.

File: src/io/ExportVTK.cpp [hunk 3]
```diff
@@ -13,23 +12,20 @@
...
-}
-
 void ExportVTK::doExport(
     const std::string &name,
     const std::string &location,
     mesh::Mesh &       mesh)
 {
   PRECICE_TRACE(name, location, mesh.getName());
   PRECICE_ASSERT(name != std::string(""));
+  PRECICE_ASSERT(!utils::MasterSlave::isParallel());
```
[the remark this answers is not in the document]
Comment (fsimonis) [reply src/io/ExportVTK.cpp:28]: My _currently undocumented_ idea is to keep the legacy vtk exporter for backwards compatibility and add `<export:vtu />` to explicitly export `vtu` files. This also neatly aligns with `<export:vtp />` from #1128.
So, the assertion really only means: _Dear developer, please don't use the `ExportVTK` (back-end) for parallel participants._

File: src/io/tests/ExportVTUTest.cpp (hunk 1 above)
Comment (fsimonis) [reply src/io/tests/ExportVTUTest.cpp:37]: I have to admit that I simply copied this testcase from the `ExportVTK` tests. I'll look into them.

File: src/io/ExportVTK.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (fsimonis) [reply src/io/ExportVTK.cpp]: Added in afb0a3e

File: src/io/Export.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (fsimonis) [reply src/io/Export.hpp]: Added in b5fafee

**Pull request merged**